Closes #71: Edit tasks (Works for both current and completed tasks)

oeng099 opened the pull request:
Implemented edit functionality on the single task scenes, with their corresponding task scene having a save task button rather than the add task button when adding tasks. Tasks are now able to have their attributes edited and for these edits to be saved.

oeng099 committed:
Commented out testSetisUnlocked unit test

oeng099 committed:
Implemented editTask method and setEditTaskScreen to have a task scene show up with existing attributes for the current task already filled in pop up

oeng099 committed:
Added a saveTask button to the Task scene which is initially disabled and not visible

oeng099 committed:
Save task button now appears when edit button for a task is clicked, replacing the add task button in the task scene

oeng099 committed:
Renamed setEditTaskScreen method to setEditTaskScene

oeng099 committed:
Implemented saveTask method in taskController to be able to save the edited task (currently only for current tasks)

oeng099 committed:
Fixed issue where the incorrect task was being moved to completed task checkbox is checked

oeng099 committed:
Fixed saveTask method to work with tasks in both current and completed sections

WiktorTumilowicz reviewed (changes requested):
I have reviewed your changes and everything functions correctly so good job!
However,
The .metadata folder is not relevant to the project and it should not be in your commits.
Please remove this folder and it's contents before merging (I suggest you add it to .gitignore).

oeng099 committed:
Removed metadata files from pull request and added .metadata to gitignore file

oeng099 commented:
> I have reviewed your changes and everything functions correctly so good job! However, The .metadata folder is not relevant to the project and it should not be in your commits. Please remove this folder and it's contents before merging (I suggest you add it to .gitignore).

I have added .metadata to the gitignore file so that those files are now removed from being committed.

WiktorTumilowicz reviewed (approved):
Good work. Should be ready to merge now (use squash and merge as reccomended by our tutor).
